### ISSUE TITLE
Push to other viewcontroller from BSImagePicker 

### ISSUE THREAD

**Issue by cliff357** [mikaoj/BSImagePicker#260]
Sorry, still have some question on #259, 

How can i push to other viewcontroller from BSImagePicker

**Comment by mikaoj** [comment]
It was supposed to work like that, if it doesn't it's a bug that will have to be looked into.
Feel free to do so :)

**Comment by mikaoj** [comment]
The problem can likely be fixed in ZoomTransitionDelegate.swift by adding a check that to/from view controllers are the expected AssetsViewController/PreviewViewController. If not don't try to apply the expand/shring animation.

**Comment by cliff357** [comment]
Thanks for quick reply, but how can I fix it? 
Would you mind to help update?

**Comment by cliff357** [comment]
Nah, I am now watching ZoomTransitionDelegate, hope I can do something on this issue :D

**Comment by mikaoj** [comment]
1. Fork this repo.
2. Clone your fork to your computer.
3. Open BSImagePicker.xcworkspace.
4. Do what I said in my previous message.
5. Test that it works as expected.
6. Commit.
7. Push.
8. Create a pull-request from your fork to this repo.

Have a look at ZoomTransitionDelegate.swift
Specifically this part:
```
extension ZoomTransitionDelegate: UINavigationControllerDelegate {
    public func navigationController(_ navigationController: UINavigationController, animationControllerFor operation: UINavigationController.Operation, from fromVC: UIViewController, to toVC: UIViewController) -> UIViewControllerAnimatedTransitioning? {
        switch operation {
        case .none:
            return nil
        case .push:
            return expandAnimator
        case .pop:
            return shrinkAnimator
        @unknown default:
            return nil
        }
    }
    
    public func navigationController(_ navigationController: UINavigationController, interactionControllerFor animationController: UIViewControllerAnimatedTransitioning) -> UIViewControllerInteractiveTransitioning? {
        if animationController === expandAnimator {
            return interactionController
        } else {
            return nil
        }
    }
}
```

I can probably fix it sometime during the week - but if you or anyone want some easy open-source karma this is up for grabs until then :)

**Comment by cliff357** [comment]
woooooooooow, thank you very much

**Comment by cliff357** [comment]
I will try your suggestion, but I do not really trust my understanding of Transition, I think i better wait for your fix haha

**Comment by mikaoj** [comment]
Something like this _might_ work. With reservation for syntax errors and typos. Just need to make sure that that expand/shrink animation still work with images, videos and live images. And also that what you want to do - push another view controller when pressing done works.
```
extension ZoomTransitionDelegate: UINavigationControllerDelegate {
    public func navigationController(_ navigationController: UINavigationController, animationControllerFor operation: UINavigationController.Operation, from fromVC: UIViewController, to toVC: UIViewController) -> UIViewControllerAnimatedTransitioning? {
        if operation == .push, toVC is PreviewViewController {
            return expandAnimator
        } else if operation == .pop, fromVC is PreviewViewController {
            return shrinkAnimator
        } else {
            return nil
        }
    }
    
    public func navigationController(_ navigationController: UINavigationController, interactionControllerFor animationController: UIViewControllerAnimatedTransitioning) -> UIViewControllerInteractiveTransitioning? {
        if animationController === expandAnimator {
            return interactionController
        } else {
            return nil
        }
    }
}
```

**Comment by cliff357** [comment]
Thanks 
But may I ask how to make sure the animation still works with images, videos, and live images? 

Sorry for my bad English.

**Comment by mikaoj** [comment]
Run the example app on your phone.
Long press on an image - is there a nice animation that zooms the image in/out? Great it works!
Do the same for videos and live photos :)

**Comment by cliff357** [comment]
ohhhhhhhhhhhhhhh, thank you very much 
tried this fix, success to push but there are some issue,

![2020-04-29-35731](https://user-images.githubusercontent.com/22883449/80573638-e39df800-8a32-11ea-9b4f-2025c788bc2c.gif)

**Comment by mikaoj** [comment]
Should be fixed in version 3.1.6 - try it out! :)
